src/Symfony/Component/Serializer/Normalizer/PropertyAccessNormalizer.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Serializer\Exception\CircularReferenceException;
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactory;
+
+/**
+ * Converts between objects and arrays using the PropertyAccess component.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PropertyAccessNormalizer extends AbstractNormalizer
+{
+    /**
+     * @var PropertyAccessorInterface
+     */
+    protected $propertyAccessor;
+
+    public function __construct(ClassMetadataFactory $classMetadataFactory = null, NameConverterInterface $nameConverter = null, PropertyAccessorInterface $propertyAccessor = null)
+    {
+        parent::__construct($classMetadataFactory, $nameConverter);
+
+        $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsNormalization($data, $format = null)
+    {
+        return is_object($data);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CircularReferenceException
+     */
+    public function normalize($object, $format = null, array $context = array())
+    {
+        if ($this->isCircularReference($object, $context)) {
+            return $this->handleCircularReference($object);
+        }
+
+        $data = array();
+        $attributes = $this->getAllowedAttributes($object, $context);
+
+        // If not using groups, detect manually
+        if (false === $attributes) {
+            $attributes = array();
+
+            // methods
+            $reflClass = new \ReflectionClass($object);
+            foreach ($reflClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflMethod) {
+                if (0 === $reflMethod->getNumberOfRequiredParameters()) {
+                    // getsetters (jQuery style, e.g. read: last(), write: last($item))
+                    $name = $reflMethod->getName();
+
+                    if (strpos($name, 'get') === 0 || strpos($name, 'has') === 0) {
+                        // getters and hassers
+                        $name = lcfirst(substr($name, 3));
+                    } elseif (strpos($name, 'is') === 0) {
+                        // issers
+                        $name = lcfirst(substr($name, 2));
+                    }
+
+                    if (!in_array($name, $attributes)) {
```

Comment:
Ok, are you sure about the performance boost (it's a real question, I've no idea of how it works internally). And what about an `array_unique` call at the end?